auslib/db.py

```diff
@@ -789,7 +789,22 @@ def getRulesMatchingQuery(self, updateQuery, fallbackChannel, transaction=None):
             if not self._localeMatchesRule(rule['locale'], updateQuery['locale']):
                 self.log.debug("%s doesn't match %s", rule['locale'], updateQuery['locale'])
                 continue
+            # If a rule has a whitelist attached to it, the rule is only
+            # considered "matching" if it passes the whitelist check.
+            # The decision about matching or not is delegated to the whitelist blob.
+            if rule.get("whitelist"):
+                self.log.debug("Matching rule requires a whitelist")
+                try:
+                    whitelist = dbo.releases.getReleaseBlob(name=rule["whitelist"], transaction=transaction)
+                    if whitelist and not whitelist.shouldServeUpdate(updateQuery):
+                        continue
+                # It shouldn't be possible for the whitelist blob not to exist,
+                # but just in case...
```

Comment:
does shouldServeUpdate exist for all blob types? could that ever raise an exception we shoudl handle here?